railgun/test/integration/knative_test.go

```diff
@@ -0,0 +1,286 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	types "k8s.io/apimachinery/pkg/types"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/kind"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	knativeCrds = "https://github.com/knative/serving/releases/download/v0.13.0/serving-crds.yaml"
+	knativeCore = "https://github.com/knative/serving/releases/download/v0.13.0/serving-core.yaml"
+)
```

Comment:
These lines wont necessarily age all that well, but I understand it's functional :thinking: 

Perhaps we can add knative deployment to the KTF with variable versioning, and provide toplevel defaults?

Reply:
Yeah, agreed, installation of Knative itself should probably live in KTF since it's a separate non-KIC app that resides in the cluster. KIC tests would then handle only creating the Knative Service resources they need for the test, or we'd add a stock general-purpose Knative Service in KTF similar to the other stuff in https://github.com/Kong/kubernetes-testing-framework/tree/main/pkg/generators/k8s. The latter may be more appropriate since Knative features would be tied to the Knative version, and that'd be determined by KTF.

The CRD/core install used here is what we currently suggest [in our docs](https://docs.konghq.com/kubernetes-ingress-controller/1.3.x/guides/using-kong-with-knative/#install-knative), but it's a bit dated at this point--Knative is now a number of versions beyond 0.13, and they've since introduced [an Operator](https://knative.dev/docs/install/knative-with-operators/) that's probably now the recommended way of managing Serving and CRD installation.

That said, our goal here is more to get existing 1.x Knative functionality into 2.x, so it may be prudent to just stick with the flat 0.13 manifests for now. 0.14 was the first version the Operator was available for, and it does look like the [0.14 core](https://github.com/knative/serving/releases/tag/v0.14.0) has sort-of breaking changes--they have a migration Job for the new StorageVersion, but we probably want to upgrade our static CRs once rather than installing the older versions and migrating them each time. I'd also expect that the first release of the Operator is probably fairly buggy, so we'd be better-served by going all the way to 0.23 if we want to migrate onto it.

Reply:
I have been understanding KTF mainly apply to test, test environment configuration, perf test etc. Integration tests have many unstable dependencies.

Reply:
In the long time, I think it is a good idea to move integration stuff into KTF.